diff --git a/utils/clone.py b/utils/clone.py
index f471f482..e80b5fd6 100755
--- a/utils/clone.py
+++ b/utils/clone.py
@@ -87,8 +87,7 @@ def clone(args): # pylint: disable=too-many-branches, too-many-locals, too-many-
     ucstaging.mkdir(exist_ok=True)
 
     get_logger().info('Cloning depot_tools')
-    dt_commit = re.search(r"depot_tools\.git'\s*\+\s*'@'\s*\+\s*'([^']+)',",
-                          Path(args.output / 'DEPS').read_text(encoding=ENCODING)).group(1)
+    dt_commit = '5e9883513113f4c6b74e33d80c066075d0fd3960'
     if not dt_commit:
         get_logger().error('Unable to obtain commit for depot_tools checkout')
         sys.exit(1)
