--- a/utils/clone.py
+++ b/utils/clone.py
@@ -87,6 +87,12 @@ def clone(args): # pylint: disable=too-many-branches, too-many-locals, too-many-
     ucstaging.mkdir(exist_ok=True)
 
     get_logger().info('Cloning depot_tools')
+    match = re.search(r"depot_tools\.git'\s*\+\s*'@'\s*\+\s*'([^']+)',",
+                      Path(args.output / 'DEPS').read_text(encoding=ENCODING))
+    if not match:
+        get_logger().error('Unable to obtain commit for depot_tools checkout')
+        sys.exit(1)
+    dt_commit = match.group(1)
     if not dtpath.exists():
         dtpath.mkdir()
         run(['git', 'init', '-q'], cwd=dtpath, check=True)
@@ -96,8 +102,8 @@ def clone(args): # pylint: disable=too-many-branches, too-many-locals, too-many-
         ],
             cwd=dtpath,
             check=True)
-    run(['git', 'fetch', '--depth=1', 'origin', 'main'], cwd=dtpath, check=True)
-    run(['git', 'reset', '--hard', 'FETCH_HEAD'], cwd=dtpath, check=True)
+    run(['git', 'fetch', '--depth=1', 'origin', dt_commit], cwd=dtpath, check=True)
+    run(['git', 'reset', '--hard', dt_commit], cwd=dtpath, check=True)
     run(['git', 'clean', '-ffdx'], cwd=dtpath, check=True)
     if iswin:
         (dtpath / 'git.bat').write_text('git')
