--- a/chrome/browser/ui/browser_commands.cc
+++ b/chrome/browser/ui/browser_commands.cc
@@ -1513,44 +1513,7 @@ void CloseTab(BrowserWindowInterface* browser) {
   }
 #endif
 
-  ToastController* toast_controller = browser->GetFeatures().toast_controller();
-  if (!toast_controller) {
-    CloseSelectedTabAndRecordTabCountMetric(browser);
-    return;
-  }
-
-  const auto& selected_tabs =
-      browser->GetTabStripModel()->selection_model().selected_tabs();
-  bool all_selected_tabs_pinned = !selected_tabs.empty();
-  for (const tabs::TabInterface* tab : selected_tabs) {
-    if (!tab->IsPinned()) {
-      all_selected_tabs_pinned = false;
-      break;
-    }
-  }
-
-  // If all selected tabs are pinned, show a confirmation toast if one isn't
-  // already shown.
-  if (all_selected_tabs_pinned &&
-      toast_controller->GetCurrentToastId() != ToastId::kClosePinnedTab) {
-    BrowserView* browser_view = BrowserView::GetBrowserViewForBrowser(browser);
-    CHECK(browser_view);
-    ui::Accelerator accelerator;
-    CHECK(
-        browser_view->GetAcceleratorForCommandId(IDC_CLOSE_TAB, &accelerator));
-
-    ToastParams params(ToastId::kClosePinnedTab);
-    params.body_string_replacement_params.emplace_back(
-        accelerator.GetShortcutText());
-    params.body_string_cardinality_param = selected_tabs.size();
-    toast_controller->MaybeShowToast(std::move(params));
-  } else {
-    CloseSelectedTabAndRecordTabCountMetric(browser);
-    if (all_selected_tabs_pinned) {
-      base::RecordAction(
-          UserMetricsAction("Tab.PinnedTabToastClosedAfterConfirmation"));
-    }
-  }
+  CloseSelectedTabAndRecordTabCountMetric(browser);
 }
 
 bool CanZoomIn(content::WebContents* contents) {
