--- a/components/embedder_support/content_settings_utils.cc
+++ b/components/embedder_support/content_settings_utils.cc
@@ -169,6 +169,11 @@ content::AllowServiceWorkerResult AllowServiceWorker(
     allow_cookies = true;
   }
 
+  if (!allow_cookies && (!top_frame_origin.has_value()
+        || !top_frame_origin->IsSameOriginWith(scope))) {
+    return content::AllowServiceWorkerResult::No();
+  }
+
   return content::AllowServiceWorkerResult::FromPolicy(!allow_javascript,
                                                        !allow_cookies);
 }
@@ -215,6 +220,11 @@ bool AllowSharedWorker(
     allow = true;
   }
 
+  if (!allow && (!top_frame_origin.has_value()
+        || !top_frame_origin->IsSameOriginWith(worker_url))) {
+    return false;
+  }
+
   content_settings::PageSpecificContentSettings::SharedWorkerAccessed(
       render_process_id, render_frame_id, worker_url, name, storage_key,
       same_site_cookies, !allow);
