--- a/chrome/browser/device_api/device_service_impl.cc
+++ b/chrome/browser/device_api/device_service_impl.cc
@@ -127,34 +127,9 @@
 
 bool IsTrustedContext(content::RenderFrameHost& host,
                       const url::Origin& origin) {
-  // Do not create the service for incognito or enterprise isolated mode.
-  if (GetProfile(host)->IsPrimaryOTRProfileWithRegularParent()) {
-    return false;
-  }
-
-#if BUILDFLAG(IS_CHROMEOS)
-  if (IsRunningInAppMode()) {
-    if (base::FeatureList::IsEnabled(
-            permissions::features::
-                kAllowMultipleOriginsForWebKioskPermissions)) {
-      return IsEqualToKioskOrigin(origin) ||
-             IsWebKioskOriginAllowed(GetProfile(host)->GetPrefs(),
-                                     origin.GetURL());
-    }
-
-    return IsEqualToKioskOrigin(origin);
-  }
-#endif  // BUILDFLAG(IS_CHROMEOS)
-
-  ASSIGN_OR_RETURN(const web_app::WebAppRegistrar& registrar,
-                   GetRegistrar(host), [] { return false; });
-  return registrar
-      .FindBestAppWithUrlInScope(
-          origin.GetURL(),
-          web_app::WebAppFilter::PolicyInstalledIsolatedWebApp() |
-              web_app::WebAppFilter::IsDevModeIsolatedApp(),
-          {.exclude_scope_extensions = true})
-      .has_value();
+  (void)host;
+  (void)origin;
+  return false;
 }
 
 bool IsAllowedByPermissionsPolicy(content::RenderFrameHost& host) {
@@ -311,16 +286,6 @@
 void DeviceServiceImpl::GetDeviceAttribute(
     void (DeviceAttributeApi::*method)(DeviceAttributeCallback callback),
     DeviceAttributeCallback callback) {
-  if (!IsAffiliatedUser(GetProfile(render_frame_host())) &&
-      !IsDevModeInstalledIwaOrigin(render_frame_host(), origin())) {
-    device_attribute_api_->ReportNotAffiliatedError(std::move(callback));
-    return;
-  }
-
-  if (!IsAllowedByContentSettings(render_frame_host(), origin())) {
-    device_attribute_api_->ReportNotAllowedError(std::move(callback));
-    return;
-  }
-
-  (device_attribute_api_.get()->*method)(std::move(callback));
-}
+  (void)method;
+  device_attribute_api_->ReportNotAllowedError(std::move(callback));
+}
