--- a/third_party/blink/renderer/platform/runtime_enabled_features.json5
+++ b/third_party/blink/renderer/platform/runtime_enabled_features.json5
@@ -1319,11 +1319,9 @@
       status: "experimental",
     },
     {
-      name: "ComputePressure",
-      status: {
-        "Android": "",
-        "default": "stable",
-      }
+      name: "ComputePressure", // disabled
+      status: "experimental",  // by default
+      public: true,
     },
     {
       // If enabled, allow concurrent animations via native paint worklets to
