--- a/third_party/blink/renderer/platform/runtime_enabled_features.json5
+++ b/third_party/blink/renderer/platform/runtime_enabled_features.json5
@@ -1432,9 +1432,7 @@
       // https://chromestatus.com/feature/5189864286978048
       // https://github.com/WICG/cpu-performance/
       name: "CpuPerformance",
-      status: "stable",
-      origin_trial_feature_name: "CpuPerformance",
-      origin_trial_allows_third_party: true,
+      status: "experimental",
     },
     {
       name: "CrashReportingStorageAPI",
