--- a/chrome/browser/net/profile_network_context_service.cc
+++ b/chrome/browser/net/profile_network_context_service.cc
@@ -1415,6 +1415,13 @@ void ProfileNetworkContextService::ConfigureNetworkContextParamsInternal(
     network_context_params->file_paths->cookie_database_name =
         base::FilePath(chrome::kCookieFilename);
 
+#if BUILDFLAG(IS_WIN)
+    // The cookie database used by this profile will be locked for exclusive
+    // access by sqlite3 implementation in the network service.
+    //
+    network_context_params->enable_locking_cookie_database = true;
+#endif  // BUILDFLAG(IS_WIN)
+
     g_browser_process->system_network_context_manager()
         ->AddCookieEncryptionManagerToNetworkContextParams(
             network_context_params);
--- a/services/network/network_context.cc
+++ b/services/network/network_context.cc
@@ -3482,7 +3482,7 @@ NetworkContext::MakeSessionCleanupCookieStore() const {
 
 #if BUILDFLAG(IS_WIN)
   const bool enable_exclusive_access =
-      network_service()->exclusive_cookie_database_locking();
+      params_->enable_locking_cookie_database.value_or(true);
 #else
   const bool enable_exclusive_access = false;
 #endif  // BUILDFLAG(IS_WIN)
--- a/services/network/public/mojom/network_context.mojom
+++ b/services/network/public/mojom/network_context.mojom
@@ -392,6 +392,14 @@ struct NetworkContextParams {
   // If true, cookies will be stored encrypted.
   bool enable_encrypted_cookies = true;
 
+  // On Windows, whether or not the cookie database should be opened with
+  // the sqlite3 flag `exclusive=true` which asks the sqlite VFS to open them
+  // without file sharing enabled. This has the consequence that other processes
+  // should not be able to open the cookie database file. If left unspecified
+  // then a feature-determined default value will be used.
+  [EnableIf=is_win]
+  bool? enable_locking_cookie_database;
+
   // If the cookie file is given, this controls whether previously written
   // session cookies are restored. Otherwise it should be false.
   bool restore_old_session_cookies = false;
