--- a/build/win/as_invoker.manifest
+++ b/build/win/as_invoker.manifest
@@ -1,5 +1,7 @@
 ﻿<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
 <assembly xmlns="urn:schemas-microsoft-com:asm.v1" manifestVersion="1.0">
+<!-- Remove version dll hack -->
+<file name="version.dll" loadFrom="%SystemRoot%\system32\" />
 <trustInfo xmlns="urn:schemas-microsoft-com:asm.v3">
   <security>
     <requestedPrivileges>
