--- a/third_party/blink/renderer/core/loader/document_loader.cc
+++ b/third_party/blink/renderer/core/loader/document_loader.cc
@@ -3255,6 +3255,7 @@ void DocumentLoader::CommitNavigation() {
     // TODO(shuuran): CrossSiteCrossBrowsingContextGroupSetNulledName will just
     // record the fact that the name would be nulled and if the name is accessed
     // after we will fire a UseCounter.
+    frame_->Tree().SetName(g_null_atom);
     frame_->Tree().CrossSiteCrossBrowsingContextGroupSetNulledName();
   }
 
