--- a/chrome/browser/user_education/user_education_service_factory.cc
+++ b/chrome/browser/user_education/user_education_service_factory.cc
@@ -122,8 +122,7 @@ bool UserEducationServiceFactory::Profil
   // In order to do user education, the browser must have a UI and not be an
   // "off-the-record" or in a demo or guest mode.
 
-  if (profile->IsIncognitoProfile() || profile->IsGuestSession() ||
-      profiles::IsDemoSession() || profiles::IsChromeAppKioskSession()) {
+  if ((true)) {
     return false;
   }
 
--- a/components/user_education/common/tutorial/tutorial_registry.cc
+++ b/components/user_education/common/tutorial/tutorial_registry.cc
@@ -25,7 +25,6 @@ bool TutorialRegistry::IsTutorialRegiste
 
 const TutorialDescription* TutorialRegistry::GetTutorialDescription(
     TutorialIdentifier id) const {
-  DCHECK(tutorial_registry_.size() > 0);
   auto pair = tutorial_registry_.find(id);
   if (pair == tutorial_registry_.end()) {
     return nullptr;
