--- a/ui/views/style/platform_style.h
+++ b/ui/views/style/platform_style.h
@@ -70,7 +70,7 @@ class VIEWS_EXPORT PlatformStyle {
   static constexpr bool kTreeViewSelectionPaintsEntireRow = BUILDFLAG(IS_MAC);
 
   // Whether ripples should be used for visual feedback on control activation.
-  static constexpr bool kUseRipples = !BUILDFLAG(IS_MAC);
+  static constexpr bool kUseRipples = false;
 
   // Whether text fields should use a "drag" cursor when not actually
   // dragging but available to do so.
--- a/ui/webui/resources/cr_elements/cr_ripple/cr_ripple_mixin.ts
+++ b/ui/webui/resources/cr_elements/cr_ripple/cr_ripple_mixin.ts
@@ -31,7 +31,7 @@ export const CrRippleMixin =
           };
         }
 
-        accessor noink: boolean = false;
+        accessor noink: boolean = true;
         rippleContainer: HTMLElement|null = null;
 
         private ripple_: CrRippleElement|null = null;
--- a/ui/webui/resources/cr_elements/cr_ripple/cr_ripple_mixin_polymer.ts
+++ b/ui/webui/resources/cr_elements/cr_ripple/cr_ripple_mixin_polymer.ts
@@ -31,7 +31,7 @@ export const CrRippleMixinPolymer =
             noink: {
               type: Boolean,
               observer: 'onNoinkChanged_',
-              value: false,
+              value: true,
             },
           };
         }
--- a/ui/webui/resources/cr_elements/cr_ripple/cr_ripple.ts
+++ b/ui/webui/resources/cr_elements/cr_ripple/cr_ripple.ts
@@ -38,7 +38,7 @@ export class CrRippleElement extends CrL
 
   accessor holdDown: boolean = false;
   accessor recenters: boolean = false;
-  accessor noink: boolean = false;
+  accessor noink: boolean = true;
 
   private ripples_: Element[] = [];
   private eventTracker_: EventTracker = new EventTracker();
