--- a/content/public/common/content_features.cc
+++ b/content/public/common/content_features.cc
@@ -428,7 +428,7 @@ BASE_FEATURE(kFedCmDelegation, base::FEA
 // flag in RuntimeEnabledFeatures on the blink side. See also the use of
 // kSetOnlyIfOverridden in content/child/runtime_features.cc. We enable it here
 // by default to support use in origin trials.
-BASE_FEATURE(kEmailVerificationProtocol, base::FEATURE_ENABLED_BY_DEFAULT);
+BASE_FEATURE(kEmailVerificationProtocol, base::FEATURE_DISABLED_BY_DEFAULT);
 
 // Enforce same-origin check for dedicated worker script URLs.
 // See https://crbug.com/496253755.
@@ -514,11 +514,11 @@ BASE_FEATURE(kForwardMemoryPressureToBli
              base::FEATURE_ENABLED_BY_DEFAULT);
 
 // Enables the Digital Credential API.
-BASE_FEATURE(kWebIdentityDigitalCredentials, base::FEATURE_ENABLED_BY_DEFAULT);
+BASE_FEATURE(kWebIdentityDigitalCredentials, base::FEATURE_DISABLED_BY_DEFAULT);
 
 // Enables the Digital Credentials Creation API.
 BASE_FEATURE(kWebIdentityDigitalCredentialsCreation,
-             base::FEATURE_ENABLED_BY_DEFAULT);
+             base::FEATURE_DISABLED_BY_DEFAULT);
 
 // Enables scrollers inside Blink to store scroll offsets in fractional
 // floating-point numbers rather than truncating to integers.
@@ -1101,7 +1101,7 @@ BASE_FEATURE(kUserMediaCaptureOnFocus, b
 
 // This is intended as a kill switch for the WebOTP Service feature. To enable
 // this feature, the experimental web platform features flag should be set.
-BASE_FEATURE(kWebOTP, base::FEATURE_ENABLED_BY_DEFAULT);
+BASE_FEATURE(kWebOTP, base::FEATURE_DISABLED_BY_DEFAULT);
 
 // Trial to disable synchronous draw for synchronous compositor (ie Android
 // WebView).
--- a/content/child/runtime_features.cc
+++ b/content/child/runtime_features.cc
@@ -495,6 +495,10 @@ void SetCustomizedRuntimeFeaturesFromCom
 // Ensures that the various ways of enabling/disabling features do not produce
 // an invalid configuration.
 void ResolveInvalidConfigurations() {
+  if (!base::FeatureList::IsEnabled(features::kWebOTP)) {
+    WebRuntimeFeatures::EnableWebOTP(false);
+  }
+
   // Fenced frames cannot be enabled without the support of the
   // browser process.
   if ((base::FeatureList::IsEnabled(features::kPrivacySandboxAdsAPIsOverride) ||
