--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -985,7 +985,10 @@ config("compiler") {
     # We use clang-rt sanitizer runtimes.
     "-Zexternal-clangrt",
 
+    "-Ctarget-cpu=x86-64-v3",
     "-Cllvm-args=-fp-contract=fast",
+    "-Ztune-cpu=generic",
+    "-Ctarget-feature=-fast-gather,+prefer-no-gather,+prefer-no-scatter",
   ]
 
   if (!is_win || force_rustc_color_output) {
@@ -1284,6 +1286,7 @@ config("thinlto_optimize_default") {
         "/prefetch-inputs",
         "/DEBUG:GHASH",
         "-mllvm:-adce-remove-loops=true",
+        "/guard:ehcont",
       ]
     } else {
       ldflags = [ "-Wl,--lto-O" + lto_opt_level ]
@@ -1348,5 +1351,7 @@ config("thinlto_optimize_max") {
         "-mllvm:-phi-node-folding-threshold=4",
         "-mllvm:-sink-insts-to-avoid-spills",
+        "-mllvm:-x86-cmov-converter",
+        "/guard:ehcont",
       ]
     } else {
       ldflags = [ "-Wl,--lto-O" + lto_opt_level ]
@@ -1361,5 +1366,6 @@ config("thinlto_optimize_max") {
         "-Wl,-mllvm,-phi-node-folding-threshold=4",
         "-Wl,-mllvm,-sink-insts-to-avoid-spills",
+        "-Wl,-mllvm,-x86-cmov-converter",
         "-Wl,-z,pack-relative-relocs",
       ]
     }
--- a/build/config/win/BUILD.gn
+++ b/build/config/win/BUILD.gn
@@ -122,7 +122,31 @@ config("compiler") {
     # microarchitecture. MSVC only supports a subset of architectures, and the
     # next step after SSE2 will be AVX.
-    if (current_cpu == "x86" || current_cpu == "x64") {
+    if (current_cpu == "x64") {
+      cflags += [
+        "-mtune=generic",
+        "-msse3",
+        "-mssse3",
+        "-msse4.1",
+        "-msse4.2",
+        "-mavx",
+        "-mavx2",
+        "-mfma",
+        "-mf16c",
+        "-mlzcnt",
+        "-mbmi",
+        "-mbmi2",
+        "-mpopcnt",
+        "-mmovbe",
+        "-mcx16",
+        "-msahf",
+        "-mpclmul",
+        "-mxsave",
+        "-mxsaveopt",
+        "-mno-gather",
+        "-mno-scatter",
+      ]
+    } else if (current_cpu == "x86") {
       cflags += [ "-msse3" ]
     }
 
     # Enable ANSI escape codes if something emulating them is around (cmd.exe
