--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -980,7 +980,10 @@ config("compiler") {
     # We use clang-rt sanitizer runtimes.
     "-Zexternal-clangrt",
 
+    "-Ctarget-cpu=x86-64-v3",
     "-Cllvm-args=-fp-contract=fast",
+    "-Ztune-cpu=skylake",
+    "-Ctarget-feature=-fast-gather,+prefer-no-gather,+prefer-no-scatter",
   ]
 
   if (!is_win || force_rustc_color_output) {
