--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -985,7 +985,9 @@ config("compiler") {
     # We use clang-rt sanitizer runtimes.
     "-Zexternal-clangrt",
 
+    "-Ctarget-cpu=x86-64-v3",
     "-Cllvm-args=-fp-contract=fast",
+    "-Ctarget-feature=-fast-gather,+prefer-no-gather,+prefer-no-scatter",
   ]
 
   if (!is_win || force_rustc_color_output) {
@@ -1284,6 +1286,7 @@ config("thinlto_optimize_default") {
         "/prefetch-inputs",
         "/DEBUG:GHASH",
         "-mllvm:-adce-remove-loops=true",
+        "/guard:ehcont",
       ]
     } else {
       ldflags = [ "-Wl,--lto-O" + lto_opt_level ]
@@ -1348,6 +1351,8 @@ config("thinlto_optimize_max") {
         "-mllvm:-phi-node-folding-threshold=4",
         "-mllvm:-sink-insts-to-avoid-spills",
         "-mllvm:-spec-exec-max-speculation-cost=15",
+        "-mllvm:-x86-cmov-converter",
+        "/guard:ehcont",
       ]
     } else {
       ldflags = [ "-Wl,--lto-O" + lto_opt_level ]
@@ -1361,6 +1366,7 @@ config("thinlto_optimize_max") {
         "-Wl,-mllvm,-phi-node-folding-threshold=4",
         "-Wl,-mllvm,-sink-insts-to-avoid-spills",
         "-Wl,-mllvm,-spec-exec-max-speculation-cost=15",
+        "-Wl,-mllvm,-x86-cmov-converter",
         "-Wl,-z,pack-relative-relocs",
       ]
     }
--- a/build/config/win/BUILD.gn
+++ b/build/config/win/BUILD.gn
@@ -122,7 +122,12 @@ config("compiler") {
     # microarchitecture. MSVC only supports a subset of architectures, and the
     # next step after SSE2 will be AVX.
     if (current_cpu == "x86" || current_cpu == "x64") {
-      cflags += [ "-msse3" ]
+      cflags += [
+        "-march=x86-64-v3",
+        "-mxsaveopt",
+        "-mno-gather",
+        "-mno-scatter",
+      ]
     }
 
     # Enable ANSI escape codes if something emulating them is around (cmd.exe
