--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -548,10 +548,10 @@ config("compiler") {
   # Linux/Android/Fuchsia common flags setup.
   # ---------------------------------
   if (is_linux || is_chromeos || is_android || is_fuchsia) {
-    asmflags += [ "-fPIC" ]
-    cflags += [ "-fPIC" ]
-    ldflags += [ "-fPIC" ]
-    rustflags += [ "-Crelocation-model=pic" ]
+    asmflags += [ "-fPIC", "-w", ]
+    cflags += [ "-fPIC", "-w", ]
+    ldflags += [ "-fPIC", "-w", ]
+    rustflags += [ "-Crelocation-model=pic", "-Awarnings", ]
 
     if (!is_clang) {
       # Use pipes for communicating between sub-processes. Faster.
@@ -664,7 +664,7 @@ config("compiler") {
 
     # TODO(crbug.com/40192287): Investigate why/if this should be needed.
     if (is_win) {
-      cflags += [ "/clang:-ffp-contract=off" ]
+      cflags += [ "/clang:-ffp-contract=fast" ]
     } else {
       cflags += [ "-ffp-contract=off" ]
     }
@@ -993,6 +993,9 @@ config("compiler") {
 
     # We use clang-rt sanitizer runtimes.
     "-Zexternal-clangrt",
+
+     "-Ctarget-cpu=x86-64-v3",
+     "-Cllvm-args=-fp-contract=fast",
   ]
 
   if (!is_win || force_rustc_color_output) {
@@ -1265,12 +1268,18 @@ config("disallow_unstable_features") {
 # building with ThinLTO, no optimization is performed in the link step.
 config("thinlto_optimize_default") {
   if (!is_debug && use_thin_lto && is_a_target_toolchain) {
-    lto_opt_level = 0
+    if (thin_lto_enable_optimizations) {
+      lto_opt_level = 2
+    } else {
+      lto_opt_level = 0
+    }
 
     if (is_win) {
       ldflags = [ "/opt:lldlto=" + lto_opt_level ]
+      ldflags += [ "-opt:lldltocgo=3" ]
     } else {
       ldflags = [ "-Wl,--lto-O" + lto_opt_level ]
+      ldflags += [ "-Wl,--lto-CGO3" ]
     }
 
     if (toolchain_supports_rust_thin_lto) {
@@ -1303,8 +1312,10 @@ config("thinlto_optimize_max") {
 
     if (is_win) {
       ldflags = [ "/opt:lldlto=" + lto_opt_level ]
+      ldflags += [ "-opt:lldltocgo=3" ]
     } else {
       ldflags = [ "-Wl,--lto-O" + lto_opt_level ]
+      ldflags += [ "-Wl,--lto-CGO3" ]
     }
 
     if (toolchain_supports_rust_thin_lto) {
@@ -1345,7 +1356,7 @@ config("compiler_cpu_abi") {
         cflags += [ "-m64" ]
         ldflags += [ "-m64" ]
       }
-      cflags += [ "-msse3" ]
+      cflags += [ "-march=x86-64-v3", ]
       if (target_os == "chromeos" && cros_target_cpu_arch != "" &&
           is_a_target_toolchain) {
         cflags += [ "-march=$cros_target_cpu_arch" ]
@@ -2782,6 +2793,50 @@ if (is_win) {
     ]
   }
   common_optimize_on_ldflags = []
+  common_optimize_on_ldflags += [
+    "-mllvm:-memdep-block-scan-limit=1000",
+    "-mllvm:-aggressive-ext-opt",
+    "-mllvm:-costmodel-reduxcost",
+    "-mllvm:-enable-cold-section",
+    "-mllvm:-enable-cond-stores-vec",
+    "-mllvm:-enable-dfa-jump-thread",
+    "-mllvm:-enable-gvn-hoist",
+    "-mllvm:-enable-gvn-sink",
+    "-mllvm:-enable-interleaved-mem-accesses",
+    "-mllvm:-enable-ipra",
+    "-mllvm:-enable-load-pre",
+    "-mllvm:-enable-local-reassign",
+    "-mllvm:-x86-machine-combiner=true",
+    "-mllvm:-enable-masked-interleaved-mem-accesses",
+    "-mllvm:-enable-nontrivial-unswitch",
+    "-mllvm:-exhaustive-register-search",
+    "-mllvm:-force-precise-rotation-cost",
+    "-mllvm:-force-vector-interleave=4",
+    "-mllvm:-gvn-hoist-max-bbs=-1",
+    "-mllvm:-gvn-hoist-max-chain-length=-1",
+    "-mllvm:-gvn-hoist-max-depth=-1",
+    "-mllvm:-inline-enable-cost-benefit-analysis",
+    "-mllvm:-jump-threading-threshold=30",
+    "-mllvm:-rotation-max-header-size=64",
+    "-mllvm:-lsr-complexity-limit=2147483647",
+    "-mllvm:-machine-sink-split",
+    "-mllvm:-misched-cluster=true",
+    "-mllvm:-phi-node-folding-threshold=3",
+    "-mllvm:-precise-rotation-cost",
+    "-mllvm:-scalar-evolution-max-add-rec-size=1000",
+    "-mllvm:-simplifycfg-sink-common",
+    "-mllvm:-slp-vectorize-hor-store",
+    "-mllvm:-spec-exec-max-speculation-cost=10",
+    "-mllvm:-tail-merge-threshold=100",
+    "-mllvm:-unroll-allow-partial",
+    "-mllvm:-unroll-runtime-epilog",
+    "-mllvm:-vectorizer-maximize-bandwidth",
+    "-mllvm:-x86-cmov-converter",
+    "-mllvm:-x86-early-ifcvt",
+    "-mllvm:-x86-pad-for-align",
+    "-mllvm:-scheditins",
+    "-mllvm:-x86-use-vzeroupper",
+  ]
 
   # /OPT:ICF is not desirable in Debug builds, since code-folding can result in
   # misleading symbols in stack traces.
@@ -2898,10 +2959,8 @@ config("optimize") {
   if (is_win) {
     # clang-cl's /O2 corresponds to clang's -O3, and really want -O2 for
     # consistency with the other platforms.
-    cflags = [
-               "/O2",
-               "/clang:-O2",
-             ] + common_optimize_on_cflags
+    cflags = [ "-O2" ] + common_optimize_on_cflags
+    # cflags = [ "-mllvm", "-pgo-cold-func-opt=minsize", ]
 
     # The `-O3` for clang turns on extra optimizations compared to the standard
     # `-O2`. But for rust, `-Copt-level=3` is the default and is thus reliable
--- a/build/config/win/BUILD.gn
+++ b/build/config/win/BUILD.gn
@@ -120,7 +120,13 @@ config("compiler") {
     # microarchitecture. MSVC only supports a subset of architectures, and the
     # next step after SSE2 will be AVX.
     if (current_cpu == "x86" || current_cpu == "x64") {
-      cflags += [ "-msse3" ]
+      cflags += [
+        "-mtune=haswell",
+        "-msse3", "-mssse3", "-msse4.1", "-msse4.2",
+        "-mavx", "-mavx2", "-mfma", "-mf16c",
+        "-mlzcnt", "-mbmi2", "-mpopcnt", "-mmovbe",
+        "/clang:-w",
+      ]
     }
 
     # Enable ANSI escape codes if something emulating them is around (cmd.exe
@@ -190,7 +196,7 @@ config("compiler") {
     if (use_lld) {
       # String tail merging leads to smaller binaries, but they don't compress
       # as well, leading to increased mini_installer size (crbug.com/838449).
-      ldflags += [ "/OPT:NOLLDTAILMERGE" ]
+      ldflags += [ "/OPT:LLDTAILMERGE" ]
     }
 
     # TODO(siggi): Is this of any use anymore?
--- a/chrome/installer/mini_installer/mini_installer_exe_main.cc
+++ b/chrome/installer/mini_installer/mini_installer_exe_main.cc
@@ -59,7 +59,7 @@ void* memset(void* dest, int c, size_t count) {
   return dest;
 }
 
-#if defined(_DEBUG) && defined(ARCH_CPU_ARM64)
+// #if defined(_DEBUG) && defined(ARCH_CPU_ARM64)
 // The compiler generates calls to memcpy for ARM64 debug builds so we need to
 // supply a memcpy implementation in that configuration.
 // See comments above for why we do this incantation.
@@ -71,5 +71,5 @@ void* memcpy(void* destination, const void* source, size_t count) {
     *dst++ = *src++;
   return destination;
 }
-#endif
+// #endif
 }  // extern "C"
