--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -549,10 +549,10 @@ config("compiler") {
   # Linux/Android/Fuchsia common flags setup.
   # ---------------------------------
   if (is_linux || is_chromeos || is_android || is_fuchsia) {
-    asmflags += [ "-fPIC" ]
-    cflags += [ "-fPIC" ]
-    ldflags += [ "-fPIC" ]
-    rustflags += [ "-Crelocation-model=pic" ]
+    asmflags += [ "-fPIC", "-w", ]
+    cflags += [ "-fPIC", "-w", ]
+    ldflags += [ "-fPIC", "-w", ]
+    rustflags += [ "-Crelocation-model=pic", "-Awarnings", ]
 
     if (!is_clang) {
       # Use pipes for communicating between sub-processes. Faster.
@@ -665,9 +665,9 @@ config("compiler") {
 
     # TODO(crbug.com/40192287): Investigate why/if this should be needed.
     if (is_win) {
-      cflags += [ "/clang:-ffp-contract=off" ]
+      cflags += [ "/clang:-ffp-contract=fast" ]
     } else {
-      cflags += [ "-ffp-contract=off" ]
+      cflags += [ "-ffp-contract=fast" ]
     }
 
     # Enable ELF CREL (see crbug.com/357878242) for all platforms that use ELF.
@@ -994,6 +994,10 @@ config("compiler") {
 
     # We use clang-rt sanitizer runtimes.
     "-Zexternal-clangrt",
+
+     "-Ctarget-cpu=x86-64-v3",
+     "-Cllvm-args=-fp-contract=fast",
+     "-Ztune-cpu=skylake",
   ]
 
   if (!is_win || force_rustc_color_output) {
@@ -1266,13 +1270,24 @@ config("disallow_unstable_features") {
 # building with ThinLTO, no optimization is performed in the link step.
 config("thinlto_optimize_default") {
   if (!is_debug && use_thin_lto && is_a_target_toolchain) {
-    lto_opt_level = 0
+    if (thin_lto_enable_optimizations) {
+      lto_opt_level = 2
+    } else {
+      lto_opt_level = 0
+    }
 
     if (is_win) {
       ldflags = [ "/opt:lldlto=" + lto_opt_level ]
+      ldflags += [ "-opt:lldltocgo=3" ]
+      ldflags += [
+        "-mllvm:-adce-remove-loops=true",
+        "-mllvm:-x86-align-branch-boundary=32",
+        "-mllvm:-x86-align-branch=jcc+jmp+indirect",
+    ]
     } else {
       ldflags = [ "-Wl,--lto-O" + lto_opt_level ]
-    }
+      ldflags += [ "-Wl,--lto-CGO3" ]
+      }
 
     if (toolchain_supports_rust_thin_lto) {
       # We always point Rust to a linker that performs LTO, so we don't want Rust
@@ -1304,8 +1319,26 @@ config("thinlto_optimize_max") {
 
     if (is_win) {
       ldflags = [ "/opt:lldlto=" + lto_opt_level ]
+      ldflags += [ "-opt:lldltocgo=3" ]
+      ldflags += [
+        "-mllvm:-import-hot-multiplier=15",
+        "-mllvm:-adce-remove-loops=true",
+        "-mllvm:-enable-gvn-hoist",
+        "-mllvm:-enable-gvn-sink",
+        "-mllvm:-machine-sink-split",
+        "-mllvm:-phi-node-folding-threshold=4",
+        "-mllvm:-sink-insts-to-avoid-spills",
+        "-mllvm:-spec-exec-max-speculation-cost=15",
+        "-mllvm:-x86-align-branch-boundary=32",
+        "-mllvm:-x86-align-branch=jcc+jmp+indirect",
+        "-mllvm:-x86-cmov-converter",
+    ]
     } else {
       ldflags = [ "-Wl,--lto-O" + lto_opt_level ]
+      ldflags += [ "-Wl,--lto-CGO3" ]
+      ldflags += [
+        "-Wl,-mllvm,-import-hot-multiplier=15",
+      ]
     }
 
     if (toolchain_supports_rust_thin_lto) {
@@ -1346,7 +1379,7 @@ config("compiler_cpu_abi") {
         cflags += [ "-m64" ]
         ldflags += [ "-m64" ]
       }
-      cflags += [ "-msse3" ]
+      cflags += [ "-march=x86-64-v3", "-mtune=skylake", ]
       if (target_os == "chromeos" && cros_target_cpu_arch != "" &&
           is_a_target_toolchain) {
         cflags += [ "-march=$cros_target_cpu_arch" ]
@@ -2875,10 +2908,7 @@ config("optimize") {
   if (is_win) {
     # clang-cl's /O2 corresponds to clang's -O3, and really want -O2 for
     # consistency with the other platforms.
-    cflags = [
-               "/O2",
-               "/clang:-O2",
-             ] + common_optimize_on_cflags
+    cflags = [ "-O2" ] + common_optimize_on_cflags
 
     # The `-O3` for clang turns on extra optimizations compared to the standard
     # `-O2`. But for rust, `-Copt-level=3` is the default and is thus reliable
@@ -2961,6 +2991,9 @@ config("optimize_max") {
     # Favor speed over size, /O2 must be before the common flags.
     # /O2 implies /Ot, /Oi, and /GF.
     cflags = [ "/O2" ] + common_optimize_on_cflags
+    if (is_clang) {
+      cflags += [ "/clang:-O3" ]
+  }
   } else if (optimize_for_fuzzing) {
     cflags = [ "-O1" ] + common_optimize_on_cflags
   } else {
--- a/build/config/win/BUILD.gn
+++ b/build/config/win/BUILD.gn
@@ -120,7 +120,13 @@ config("compiler") {
     # microarchitecture. MSVC only supports a subset of architectures, and the
     # next step after SSE2 will be AVX.
     if (current_cpu == "x86" || current_cpu == "x64") {
-      cflags += [ "-msse3" ]
+      cflags += [
+        "-mtune=skylake",
+        "-msse3", "-mssse3", "-msse4.1", "-msse4.2",
+        "-mavx", "-mavx2", "-mfma", "-mf16c",
+        "-mlzcnt", "-mbmi", "-mbmi2", "-mpopcnt",
+        "-mmovbe", "/clang:-w",
+      ]
     }
 
     # Enable ANSI escape codes if something emulating them is around (cmd.exe
@@ -190,7 +196,7 @@ config("compiler") {
     if (use_lld) {
       # String tail merging leads to smaller binaries, but they don't compress
       # as well, leading to increased mini_installer size (crbug.com/838449).
-      ldflags += [ "/OPT:NOLLDTAILMERGE" ]
+      ldflags += [ "/OPT:LLDTAILMERGE" ]
     }
 
     # TODO(siggi): Is this of any use anymore?
--- a/chrome/installer/mini_installer/mini_installer_exe_main.cc
+++ b/chrome/installer/mini_installer/mini_installer_exe_main.cc
@@ -59,7 +59,7 @@ void* memset(void* dest, int c, size_t count) {
   return dest;
 }
 
-#if defined(_DEBUG) && defined(ARCH_CPU_ARM64)
+// #if defined(_DEBUG) && defined(ARCH_CPU_ARM64)
 // The compiler generates calls to memcpy for ARM64 debug builds so we need to
 // supply a memcpy implementation in that configuration.
 // See comments above for why we do this incantation.
@@ -71,5 +71,5 @@ void* memcpy(void* destination, const void* source, size_t count) {
     *dst++ = *src++;
   return destination;
 }
-#endif
+// #endif
 }  // extern "C"
