--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -529,10 +529,10 @@ config("compiler") {
   # Linux/Android/Fuchsia common flags setup.
   # ---------------------------------
   if (is_linux || is_chromeos || is_android || is_fuchsia) {
-    asmflags += [ "-fPIC" ]
-    cflags += [ "-fPIC" ]
-    ldflags += [ "-fPIC" ]
-    rustflags += [ "-Crelocation-model=pic" ]
+    asmflags += [ "-fPIC", "-w", ]
+    cflags += [ "-fPIC", "-w", ]
+    ldflags += [ "-fPIC", "-w", ]
+    rustflags += [ "-Crelocation-model=pic", "-Awarnings", ]
 
     if (!is_clang) {
       # Use pipes for communicating between sub-processes. Faster.
@@ -651,9 +651,9 @@ config("compiler") {
 
     # TODO(crbug.com/40192287): Investigate why/if this should be needed.
     if (is_win) {
-      cflags += [ "/clang:-ffp-contract=off" ]
+      cflags += [ "/clang:-ffp-contract=fast" ]
     } else {
-      cflags += [ "-ffp-contract=off" ]
+      cflags += [ "-ffp-contract=fast" ]
     }
 
     # Enable ELF CREL (see crbug.com/357878242) for all platforms that use ELF.
@@ -980,6 +980,8 @@ config("compiler") {
 
     # We use clang-rt sanitizer runtimes.
     "-Zexternal-clangrt",
+
+    "-Cllvm-args=-fp-contract=fast",
   ]
 
   if (!is_win || force_rustc_color_output) {
@@ -1260,12 +1262,27 @@ config("disallow_unstable_features") {
 # building with ThinLTO, no optimization is performed in the link step.
 config("thinlto_optimize_default") {
   if (!is_debug && use_thin_lto && is_a_target_toolchain) {
-    lto_opt_level = 0
+    if (thin_lto_enable_optimizations) {
+      lto_opt_level = 2
+    } else {
+      lto_opt_level = 0
+    }
 
     if (is_win) {
       ldflags = [ "/opt:lldlto=" + lto_opt_level ]
+      ldflags += [ "-opt:lldltocgo=3" ]
+      ldflags += [
+        "-mllvm:-adce-remove-loops=true",
+        "-mllvm:-polly",
+      ]
     } else {
       ldflags = [ "-Wl,--lto-O" + lto_opt_level ]
+      ldflags += [ "-Wl,--lto-CGO3" ]
+      ldflags += [
+        "-Wl,-mllvm,-adce-remove-loops=true",
+        "-Wl,-z,pack-relative-relocs",
+        "-Wl,-mllvm,-polly",
+      ]
     }
 
     if (toolchain_supports_rust_thin_lto) {
@@ -1299,8 +1316,33 @@ config("thinlto_optimize_max") {
 
     if (is_win) {
       ldflags = [ "/opt:lldlto=" + lto_opt_level ]
+      ldflags += [ "-opt:lldltocgo=3" ]
+      ldflags += [
+        "-mllvm:-import-hot-multiplier=15",
+        "-mllvm:-adce-remove-loops=true",
+        "-mllvm:-enable-gvn-hoist",
+        "-mllvm:-enable-gvn-sink",
+        "-mllvm:-machine-sink-split",
+        "-mllvm:-phi-node-folding-threshold=4",
+        "-mllvm:-sink-insts-to-avoid-spills",
+        "-mllvm:-spec-exec-max-speculation-cost=15",
+        "-mllvm:-polly",
+      ]
     } else {
       ldflags = [ "-Wl,--lto-O" + lto_opt_level ]
+      ldflags += [ "-Wl,--lto-CGO3" ]
+      ldflags += [
+        "-Wl,-mllvm,-import-hot-multiplier=15",
+        "-Wl,-mllvm,-adce-remove-loops=true",
+        "-Wl,-mllvm,-enable-gvn-hoist",
+        "-Wl,-mllvm,-enable-gvn-sink",
+        "-Wl,-mllvm,-machine-sink-split",
+        "-Wl,-mllvm,-phi-node-folding-threshold=4",
+        "-Wl,-mllvm,-sink-insts-to-avoid-spills",
+        "-Wl,-mllvm,-spec-exec-max-speculation-cost=15",
+        "-Wl,-z,pack-relative-relocs",
+        "-Wl,-mllvm,-polly",
+      ]
     }
 
     if (toolchain_supports_rust_thin_lto) {
@@ -1723,7 +1765,7 @@ config("runtime_library") {
   }
 
   if (use_safe_libcxx) {
-    defines += [ "_LIBCPP_HARDENING_MODE=_LIBCPP_HARDENING_MODE_EXTENSIVE" ]
+    defines += [ "_LIBCPP_HARDENING_MODE=_LIBCPP_HARDENING_MODE_FAST" ]
   } else {
     defines += [ "_LIBCPP_HARDENING_MODE=_LIBCPP_HARDENING_MODE_NONE" ]
   }
@@ -1958,7 +2000,10 @@ config("default_warnings") {
     if (is_win) {
       # TODO(thakis): https://crbug.com/617318
       # Currently RBE can not handle case sensitiveness for windows well.
-      cflags += [ "-Wno-nonportable-include-path" ]
+      cflags += [
+        "-Wno-deprecated-attributes",
+        "-Wno-nonportable-include-path",
+      ]
     }
 
     if (is_fuchsia) {
@@ -2642,10 +2687,13 @@ config("optimize_max") {
     # Favor speed over size, /O2 must be before the common flags.
     # /O2 implies /Ot, /Oi, and /GF.
     cflags = [ "/O2" ] + common_optimize_on_cflags
+    if (is_clang) {
+      cflags += [ "/clang:-O3" ]
+    }
   } else if (optimize_for_fuzzing) {
     cflags = [ "-O1" ] + common_optimize_on_cflags
   } else {
-    cflags = [ "-O2" ] + common_optimize_on_cflags
+    cflags = [ "-O3" ] + common_optimize_on_cflags
   }
   rustflags = [ "-Copt-level=3" ]
 }
--- a/build/config/compiler_cpu_abi.gn
+++ b/build/config/compiler_cpu_abi.gn
@@ -55,7 +55,13 @@ if ((is_posix && !is_apple) || is_fuchsia) {
       cpu_abi_cflags += [ "-m64" ]
       cpu_abi_ldflags += [ "-m64" ]
     }
-    cpu_abi_cflags += [ "-msse3" ]
+    cpu_abi_cflags += [
+      "-march=x86-64-v3",
+      "-mtune=skylake",
+      "-mno-gather",
+      "-mno-scatter",
+      "-mno-bmi2",
+    ]
     if (target_os == "chromeos" && cros_target_cpu_arch != "" &&
         is_a_target_toolchain) {
       cpu_abi_cflags += [ "-march=$cros_target_cpu_arch" ]
--- a/chrome/installer/mini_installer/mini_installer_exe_main.cc
+++ b/chrome/installer/mini_installer/mini_installer_exe_main.cc
@@ -59,7 +59,7 @@ void* memset(void* dest, int c, size_t count) {
   return dest;
 }
 
-#if defined(_DEBUG) && defined(ARCH_CPU_ARM64)
+// #if defined(_DEBUG) && defined(ARCH_CPU_ARM64)
 // The compiler generates calls to memcpy for ARM64 debug builds so we need to
 // supply a memcpy implementation in that configuration.
 // See comments above for why we do this incantation.
@@ -71,5 +71,5 @@ void* memcpy(void* destination, const void* source, size_t count) {
     *dst++ = *src++;
   return destination;
 }
-#endif
+// #endif
 }  // extern "C"
--- a/chrome/tools/build/win/create_installer_archive.py
+++ b/chrome/tools/build/win/create_installer_archive.py
@@ -83,21 +83,7 @@ def CompressUsingLZMA(build_dir,
         cmd.append('-mx1')
     else:
         cmd.extend([
-            # Flags equivalent to -mx9 (ultra) but with the bcj2 turned on (exe
-            # pre-filter). This results in a ~2.3MB decrease in installer size
-            # on a 24MB installer.
-            # Additionally, these settings reflect a 7zip 4.42 and up change in
-            # the definition of -mx9, increasting the dictionary size moving to
-            # 26bit = 64MB. This results in an additional ~3.5MB decrease.
-            # Older 7zip versions can support these settings, as these changes
-            # rely on existing functionality in the lzma format.
-            '-m0=BCJ2',
-            '-m1=LZMA:d27:fb128',
-            '-m2=LZMA:d22:fb128:mf=bt2',
-            '-m3=LZMA:d22:fb128:mf=bt2',
-            '-mb0:1',
-            '-mb0s1:2',
-            '-mb0s2:3',
+            '-m0=LZMA2:d24:fb64',
         ])
     cmd.extend([
         os.path.abspath(compressed_file),
