# Fix building GN on Windows
# Author: shiromichi on GitHub

--- a/tools/gn/build/build_win.ninja.template
+++ b/tools/gn/build/build_win.ninja.template
@@ -1,5 +1,5 @@
 rule cxx
-  command = $cxx /nologo /showIncludes /FC $includes $cflags /c $in /Fo$out
+  command = $cxx /utf-8 /nologo /showIncludes /FC $includes $cflags /c $in /Fo$out
   description = CXX $out
   deps = msvc
 
--- a/tools/gn/build/gen.py
+++ b/tools/gn/build/gen.py
@@ -588,6 +588,7 @@ def WriteGNNinja(path, platform, host, o
         '/D_SCL_SECURE_NO_DEPRECATE',
         '/D_UNICODE',
         '/D_WIN32_WINNT=0x0A00',
+        '/D_LEGACY_CODE_ASSUMES_STRING_VIEW_INCLUDES_XSTRING',
         '/FS',
         '/W4',
         '/Zi',
