--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -533,10 +533,10 @@ config("compiler") {
   # Linux/Android/Fuchsia common flags setup.
   # ---------------------------------
   if (is_linux || is_chromeos || is_android || is_fuchsia) {
-    asmflags += [ "-fPIC" ]
-    cflags += [ "-fPIC" ]
-    ldflags += [ "-fPIC" ]
-    rustflags += [ "-Crelocation-model=pic" ]
+    asmflags += [ "-fPIC", "-w", ]
+    cflags += [ "-fPIC", "-w", ]
+    ldflags += [ "-fPIC", "-w", ]
+    rustflags += [ "-Crelocation-model=pic", "-Awarnings", ]
 
     if (!is_clang) {
       # Use pipes for communicating between sub-processes. Faster.
@@ -659,9 +659,9 @@ config("compiler") {
 
     # TODO(crbug.com/40192287): Investigate why/if this should be needed.
     if (is_win) {
-      cflags += [ "/clang:-ffp-contract=off" ]
+      cflags += [ "/clang:-ffp-contract=fast" ]
     } else {
-      cflags += [ "-ffp-contract=off" ]
+      cflags += [ "-ffp-contract=fast" ]
     }
 
     # Enable ELF CREL (see crbug.com/357878242) for all platforms that use ELF.
@@ -988,6 +988,8 @@ config("compiler") {
 
     # We use clang-rt sanitizer runtimes.
     "-Zexternal-clangrt",
+
+    "-Cllvm-args=-fp-contract=fast",
   ]
 
   if (!is_win || force_rustc_color_output) {
@@ -1272,12 +1274,28 @@ config("disallow_unstable_features") {
 # building with ThinLTO, no optimization is performed in the link step.
 config("thinlto_optimize_default") {
   if (!is_debug && use_thin_lto && is_a_target_toolchain) {
-    lto_opt_level = 0
+    if (thin_lto_enable_optimizations) {
+      lto_opt_level = 2
+    } else {
+      lto_opt_level = 0
+    }
 
     if (is_win) {
       ldflags = [ "/opt:lldlto=" + lto_opt_level ]
+      ldflags += [ "-opt:lldltocgo=3" ]
+      ldflags += [
+        "/Brepro",
+        "/prefetch-inputs",
+        "/DEBUG:GHASH",
+        "-mllvm:-adce-remove-loops=true",
+      ]
     } else {
       ldflags = [ "-Wl,--lto-O" + lto_opt_level ]
+      ldflags += [ "-Wl,--lto-CGO3" ]
+      ldflags += [
+        "-Wl,-mllvm,-adce-remove-loops=true",
+        "-Wl,-z,pack-relative-relocs",
+      ]
     }
 
     if (toolchain_supports_rust_thin_lto) {
@@ -1311,8 +1329,30 @@ config("thinlto_optimize_max") {
 
     if (is_win) {
       ldflags = [ "/opt:lldlto=" + lto_opt_level ]
+      ldflags += [ "-opt:lldltocgo=3" ]
+      ldflags += [
+        "/Brepro",
+        "/prefetch-inputs",
+        "/DEBUG:GHASH",
+        "-mllvm:-adce-remove-loops=true",
+        "-mllvm:-enable-gvn-hoist",
+        "-mllvm:-enable-gvn-sink",
+        "-mllvm:-machine-sink-split",
+        "-mllvm:-phi-node-folding-threshold=4",
+        "-mllvm:-sink-insts-to-avoid-spills",
+      ]
     } else {
       ldflags = [ "-Wl,--lto-O" + lto_opt_level ]
+      ldflags += [ "-Wl,--lto-CGO3" ]
+      ldflags += [
+        "-Wl,-mllvm,-adce-remove-loops=true",
+        "-Wl,-mllvm,-enable-gvn-hoist",
+        "-Wl,-mllvm,-enable-gvn-sink",
+        "-Wl,-mllvm,-machine-sink-split",
+        "-Wl,-mllvm,-phi-node-folding-threshold=4",
+        "-Wl,-mllvm,-sink-insts-to-avoid-spills",
+        "-Wl,-z,pack-relative-relocs",
+      ]
     }
 
     if (toolchain_supports_rust_thin_lto) {
@@ -1736,7 +1779,7 @@ config("runtime_library") {
   }
 
   if (use_safe_libcxx) {
-    defines += [ "_LIBCPP_HARDENING_MODE=_LIBCPP_HARDENING_MODE_EXTENSIVE" ]
+    defines += [ "_LIBCPP_HARDENING_MODE=_LIBCPP_HARDENING_MODE_FAST" ]
   } else {
     defines += [ "_LIBCPP_HARDENING_MODE=_LIBCPP_HARDENING_MODE_NONE" ]
   }
@@ -1971,7 +2011,11 @@ config("default_warnings") {
     if (is_win) {
       # TODO(thakis): https://crbug.com/617318
       # Currently RBE can not handle case sensitiveness for windows well.
-      cflags += [ "-Wno-nonportable-include-path" ]
+      cflags += [
+        "/clang:-gcodeview-ghash",
+        "-Wno-deprecated-attributes",
+        "-Wno-nonportable-include-path",
+      ]
     }
 
     if (is_fuchsia) {
@@ -2681,10 +2734,13 @@ config("optimize_max") {
     # Favor speed over size, /O2 must be before the common flags.
     # /O2 implies /Ot, /Oi, and /GF.
     cflags = [ "/O2" ] + common_optimize_on_cflags
+    if (is_clang) {
+      cflags += [ "/clang:-O3" ]
+    }
   } else if (optimize_for_fuzzing) {
     cflags = [ "-O1" ] + common_optimize_on_cflags
   } else {
-    cflags = [ "-O2" ] + common_optimize_on_cflags
+    cflags = [ "-O3" ] + common_optimize_on_cflags
   }
   rustflags = [ "-Copt-level=3" ]
 }
